### PR TITLE
Fix incorrect logits returned in API

### PR DESCRIPTION
**Pull request by mihow**
This PR fixes incorrect logits being returned in the API by restructuring how classifier results are handled and returned. The changes address issue https://github.com/RolnickLab/ami-data-companion/issues/91 by ensuring logits are properly separated from scores in classification responses.

- Returns the actual logits from the model output rather than a copy of the softmax scores
- Introduces a new ClassifierResult dataclass to standardize classifier output format
- Add test coverage for expected logits and scores
- Fixes the test setup for all previous API tests that were passing because no images were in the tests requests.
- Fixes the test failures introduced in #93 (but were passing because of the discovered test setup issue)

### CODE REVIEW

**Comment by mihow** [comment]
Part of https://github.com/RolnickLab/antenna/issues/782
